src/interface/variable_pack.hpp

```diff
@@ -529,13 +554,14 @@ void AppendSparseBaseMap(const VariableVector<T> &vars, PackIndexMap *pvmap) {
 }
 
 template <typename T>
-void FillVarView(const VariableVector<T> &vars, bool coarse, ViewOfParArrays<T> &cv_out,
-                 ParArray1D<int> &sparse_id_out, ParArray1D<int> &vector_component_out,
-                 ParArray1D<bool> &allocated_out, PackIndexMap *pvmap) {
+void FillVarView(const VariableVector<T> &vars, int vsize, bool coarse,
+                 ViewOfParArrays<T> &cv_out, ParArray1D<int> &sparse_id_out,
+                 ParArray1D<int> &vector_component_out, ParArray1D<bool> &allocated_out,
+                 PackIndexMap *pvmap) {
```

Comment:
`vsize` is the number of variables. Since we are using a single `ViewOfParArrays` to store all topological components of a field, `cv_out` could have a size that is three times the number of variables. As a result, the old asserts could trigger even when everything is correct.